crates/loans/rpc/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright 2022 Interlay.
+// This file is part of Interlay.
+
+// Copyright 2021 Parallel Finance Developer.
+// This file is part of Parallel Finance.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+pub use pallet_loans_rpc_runtime_api::LoansApi as LoansRuntimeApi;
+
+use codec::Codec;
+use jsonrpsee::{
+    core::{async_trait, Error as JsonRpseeError, RpcResult},
+    proc_macros::rpc,
+    types::error::{CallError, ErrorCode, ErrorObject},
+};
+use primitives::{CurrencyId, Liquidity, Rate, Ratio, Shortfall};
+use sp_api::ProvideRuntimeApi;
+use sp_blockchain::HeaderBackend;
+use sp_rpc::number::NumberOrHex;
+use sp_runtime::{generic::BlockId, traits::Block as BlockT, FixedU128};
+
+#[rpc(client, server)]
+pub trait LoansApi<BlockHash, AccountId, Balance>
+where
+    Balance: Codec + Copy + TryFrom<NumberOrHex>,
+{
+    #[method(name = "loans_getCollateralLiquidity")]
+    fn get_account_liquidity(
+        &self,
+        account: AccountId,
+        at: Option<BlockHash>,
+    ) -> RpcResult<(Liquidity, Shortfall, Liquidity, Shortfall)>;
+    #[method(name = "loans_getMarketStatus")]
+    fn get_market_status(
+        &self,
+        asset_id: CurrencyId,
+        at: Option<BlockHash>,
+    ) -> RpcResult<(Rate, Rate, Rate, Ratio, NumberOrHex, NumberOrHex, FixedU128)>;
+    #[method(name = "loans_getLiquidationThresholdLiquidity")]
+    fn get_liquidation_threshold_liquidity(
+        &self,
+        account: AccountId,
+        at: Option<BlockHash>,
+    ) -> RpcResult<(Liquidity, Shortfall, Liquidity, Shortfall)>;
+}
+
+/// A struct that implements the [`LoansApi`].
+pub struct Loans<C, B> {
+    client: Arc<C>,
+    _marker: std::marker::PhantomData<B>,
+}
+
+impl<C, B> Loans<C, B> {
+    /// Create new `Loans` with the given reference to the client.
+    pub fn new(client: Arc<C>) -> Self {
+        Self {
+            client,
+            _marker: Default::default(),
+        }
+    }
+}
+
+pub enum Error {
+    RuntimeError,
+    AccountLiquidityError,
+    MarketStatusError,
+}
+
+impl From<Error> for i32 {
+    fn from(e: Error) -> i32 {
+        match e {
+            Error::RuntimeError => 1,
+            Error::AccountLiquidityError => 2,
+            Error::MarketStatusError => 3,
+        }
+    }
+}
```

Comment:
Why do we need this trait implementation?

Reply:
It was there in the original implementation, they have these custom RPC errors. As far as I can tell, serializing for json-rpc requires this trait, and since we use JsonRpsee default errors, there's no need for us to implement the trait. 

[Example](https://github.com/parallel-finance/parallel/blob/f15d2ca95450392364a72e706a137db344905687/pallets/loans/rpc/src/lib.rs#L176) in their code, [example](https://github.com/interlay/interbtc/blob/c88fc96634160ed585fb8467a98992f52b622a69/crates/issue/rpc/src/lib.rs#L27) in our code.